functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
+        logging.info('Closed the db connection')   
 
-#POST API method function       
-def post(param):
-    return ('You selected POST method with {param} values. Functionality under construction')
-            # STARTER CODE FOR NEXT SPRINT, SQL QUERY TESTED
-            #     #insert row into tasks table / create a new task
-            #     #createdDate #'20120618 10:34:09 AM' and title/description are hardcoded for sprint1, update with automated date stamp and url params later
-            #     sql_query = ("""INSERT INTO dbo.tasks (userId, title, description, createdDate)
-            #     VALUES (?, ?, ?, '20120618 10:34:09 AM')""")
-            #     cursor.execute(sql_query, userId, 'Do It', 'Almost like Nike motto')   
-            #     logging.info('Executed the query')
-
-#UPDATE API method function
-def update(param1, param2):
+#PUT API method function
+def update(param1, param2, task_req_body):
     #connects to db
     cnxn = connect()
     cursor = cnxn.cursor()
-    logging.info('opened connection')        
-    logging.info('Going to execute update query')
     try:
-        # update task: title and description. Client passes userId and taskId, 
-        # for sprint 1 other fields are HARDCODED
-        sql_query = ("""UPDATE [dbo].[tasks]
-        SET title = 'Title updated by team1', description = 'Description updated by team1'
-        WHERE userId = ? AND taskId = ?""")   
-        rowcount = cursor.execute(sql_query, param1, param2).rowcount
-        logging.info(f"Executed the query: {rowcount} rows affected")
-#commits changes to db
+        logging.info('opened connection')        
+        logging.info('Going to execute UPDATE task query')
+        # task_req_body input check
+        logging.debug("Checking the request body for necessary fields to update a task")
+        try:
+            assert 'title' in task_req_body, "New task request body did not contain field: 'title'"
+            assert 'description' in task_req_body, "New task request body did not contain field: 'description'"
+            assert 'dueDate' in task_req_body, "New user request body did not contain field: 'dueDate'"
+        except AssertionError as task_req_body_content_error:
+            logging.critical('New user request body did not contain fields to update a task')
+            return func.HttpResponse(task_req_body_content_error.args[0], status_code=400)
+        logging.info('New user request body contained necessary fields to update a task')  
+        # Unpack task data
+        title = task_req_body.get('title')
+        description = task_req_body.get('description')
+        dueDate = datetime.strptime(task_req_body.get('dueDate'), '%d/%m/%y %H:%M:%S')     
+        # update task: title and description
+        sql_query = """UPDATE [dbo].[tasks]
+        SET title = '{}', description = '{}', dueDate = '{}'
+        WHERE userId = ? AND taskId = ?""".format(title, description, dueDate)

```

Comment:
I think instead of format, title, description, etc can also use the ? format. You can also package all your params as a tuple (title, description, dueDate, etc..) and pass them as cursor.execute(sql_query, params).